instagram/src/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+const withAuthenticate = PostsPage => Login =>
+    class extends React.Component {
+        constructor(props) {
+              super(props);
```

Comment:
I would recommend thinking of ways that you could possibly pass your conditional as a prop for this as well as your username and password states. You don't generally want too much hardcoded state to live inside of your auth HOC in a real application. This however works really well for this application, great job.